src/tls.c

```diff
@@ -7178,35 +7249,77 @@ static int TLSX_KeyShare_GenOqsKey(WOLFSSL *ssl, KeyShareEntry* kse)
         return BAD_FUNC_ARG;
     }
 
-    pubKey = (byte*)XMALLOC(kem->length_public_key, ssl->heap,
-                               DYNAMIC_TYPE_PUBLIC_KEY);
-    privKey = (byte*)XMALLOC(kem->length_secret_key, ssl->heap,
-                                DYNAMIC_TYPE_PRIVATE_KEY);
-    if (pubKey == NULL || privKey == NULL) {
-        WOLFSSL_MSG("memory allocation failure");
+    ecc_kse = XMALLOC(sizeof(*ecc_kse), ssl->heap, DYNAMIC_TYPE_TLSX);
+    if (ecc_kse == NULL) {
+        WOLFSSL_MSG("ecc_kse memory allocation failure");
         ret = MEMORY_ERROR;
     }
-    else if (OQS_KEM_keypair(kem, pubKey, privKey) == OQS_SUCCESS) {
-        kse->pubKey = pubKey;
-        kse->pubKeyLen = (word32) kem->length_public_key;
-        pubKey = NULL;
 
-        kse->key = privKey;
-        kse->keyLen = (word32) kem->length_secret_key;
-        privKey = NULL;
-        ret = 0;
+    if (ret == 0) {
+        XMEMSET(ecc_kse, 0, sizeof(*ecc_kse));
     }
-    else {
-        WOLFSSL_MSG("liboqs keygen failure");
-        ret = BAD_FUNC_ARG;
+
+    if (ret == 0 && ecc_group != 0) {
+        ecc_kse->group = ecc_group;
+        ret = TLSX_KeyShare_GenEccKey(ssl, ecc_kse);
+        if (ret != 0) {
+            /* No message, TLSX_KeyShare_GenEccKey() will do it. */
+            return ret;
```

Comment:
Will this leak `ecc_kse`?

Reply:
Yes, it will.  Fix is done. Please see next commit.